#161255318 Users should be able to view and edit their profiles

njeri-ngigi opened the pull request:
#### What does this PR do?
- Adds a profile page
- Adds functionality to allow a user to edit their profile
#### Description of tasks to be completed
> Users should be able to view and edit their profiles

**Why this is important**
- Users should be able to ammend/edit their profiles depending on change of preference to the full disclosure of the individual user
_**As a User**_
- I want to have access to my profile.
- I want to be able to edit my profile to my preference.
#### Relevant PT stories
[Users should be able to view and edit their profiles](https://www.pivotaltracker.com/story/show/161255318)

#### Screenshots:
![image](https://user-images.githubusercontent.com/28973383/49668356-e3344400-fa6e-11e8-8bd0-38cb28aeffb9.png)
![image](https://user-images.githubusercontent.com/28973383/49668359-e6c7cb00-fa6e-11e8-9347-6251e3b54636.png)
![image](https://user-images.githubusercontent.com/28973383/49668364-e9c2bb80-fa6e-11e8-950a-58101b3dbf5c.png)

Commit:
(Feature #161255318): Edit Profile:
- add components: modal(with edit profile form), user bio, user stories, followers, following
- add redux reducers and actions
- fetch data using axios from API
[Feature #161255318]

Commit:
(Feature #161255318): Edit Profile:
- add components: modal(with edit profile form), user bio, user stories, followers, following
- add redux reducers and actions
- fetch data using axios from API
[Feature #161255318]

Commit:
(Feature #161255318): Edit Profile:
- Add tests for actions
- Adjust css styling to make page responsive
- Add edit profile picture
[Feature #161255318]

Commit:
(Feature #161255318): Edit Profile:
- Resolve merge conflicts from rebase
[Feature #161255318]

Commit:
(Feature #161255318): Edit Profile:
- Resolve merge conflicts from rebase
[Feature #161255318]

meshnesh commented:
@njeri-ngigi your P.R has code climate issues of code duplication, please fix this so that the P.R can be merged, https://codeclimate.com/github/andela/ah-codeofduty-frontend/pull/13

maxwellgithinji reviewed (approved):
LGTM

njeri-ngigi committed:
(Feature #162516734): Authentication:
- Refactor input elements to have one root called
- Refactor similar code in tests
[Feature #162516734]

njeri-ngigi committed:
(Feature #162516734): Authentication:
- Refactor profile actions getFollowers and getFollowing into one method
- Refactor profile actions editProfile and cancelEdit into one method
[Feature #162516734]

njeri-ngigi committed:
(Feature #162516734): Authentication:
- Update action tests to reflect refactoring changes
[Feature #162516734]

njeri-ngigi committed:
(Feature #161255318): User Profile:
- Refactor bio.jsx to remove repeated bootstrap columns
[Feature #161255318]

njeri-ngigi committed:
(Feature #161255318): User Profile:
- Refactor actions.test.js to remove repeated code
[Feature #161255318]

njeri-ngigi committed:
(Feature #161255318): User Profile:
- Refactor actions.test.js to remove repeated code
[Feature #161255318]

njeri-ngigi committed:
(Feature #161255318): User Profile:
- Add tests to cover profile reducers
[Feature #161255318]
